### DISCUSSION TITLE
Specific types for router and Link

### DISCUSSION THREAD

**Discussion by gamegee** [vercel/next.js#18082] [Help]
I was looking at the docs, trying to find a way to get strict typing with nextjs router.

Imagine you have the following routes:
- '/about'
- '/user'

Then with strict typing I would expect:
```tsx
// Good - ts pass
router.push({
   pathname: '/about'
})

// BAD - ts error because this route does not exist
router.push({
   pathname: '/nextjs'
})

// Same goes for Link
<Link href={{
  pathname: '/about' // pass
}}>
```

Is there a way to achieve that? Does the library export something such as list of routes or routes config ?
Would be nice to have such typescript support for better routes scale and maintainability.

**Comment by solufa** [comment]
Pathpida may be the solution.

```tsx
import Link from 'next/link'
import { pagesPath } from '../lib/$path' // Automatically generated by pathpida

export default () => {
  const onClick = useCallback(() => {
    router.push(pagesPath.post._pid(1).$url())
  }, [])

  return <>
    <Link href={pagesPath.post._slug(['a', 'b', 'c']).$url()} />
    <div onClick={onClick} />
  </>
}
```

https://github.com/aspida/pathpida

Our Japanese team has been maintaining it for over a year and can answer your questions!
We've submitted it to ProductHunt, and I hope you'll vote for it!

https://www.producthunt.com/posts/pathpida
